File: remember-everything/src/main/java/com/niuma/remembereverything/controller/UserController.java
package com.niuma.remembereverything.controller;

import com.niuma.remembereverything.entity.User;
import java.util.List;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.security.core.session.SessionRegistry;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping(value = "/management/user")
@RequiredArgsConstructor
@Slf4j
public class UserController {

  private final SessionRegistry sessionRegistry;

  @GetMapping("/online-list")
  public List<User> users() {
    List<Object> principals = sessionRegistry.getAllPrincipals();
    log.info("online user`s amount = {}", principals.size());
    Object principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
    System.out.println(principal);
    return principals.stream()
        .map(User.class::cast)
        .toList();
  }


}

File: src/main/kotlin/com/niuma/remembereverythingapp/web/UserController.kt
package com.niuma.remembereverythingapp.web

import com.niuma.remembereverythingapp.base.exception.BusinessException
import com.niuma.remembereverythingapp.base.response.ResultCode
import com.niuma.remembereverythingapp.entity.User
import com.niuma.remembereverythingapp.repository.UserRepository
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController


@RestController
@RequestMapping("/api")
class UserController(
  private val userRepository: UserRepository,
) {

  @GetMapping("/me")
  fun currentUser(@AuthenticationPrincipal user: User): User {
    throw BusinessException(ResultCode.INTERNAL_SERVER_ERROR.code, ResultCode.INTERNAL_SERVER_ERROR.message)
  }

  @GetMapping("users")
  fun users(): List<User> {
    return userRepository.findAll()
  }
}